File: src/api/Bysykkel.Application/Models/Bysykkel/Station.cs
namespace Bysykkel.Application.Models.Bysykkel;

public class Station
{
    public Station(StationInfo info, StationStatus status)
    {
        Info = info;
        Status = status;
    }
    public StationInfo Info { get; set; }
    public StationStatus Status { get; set; }
}

File: src/client/src/types/station.ts
export type VehicleType = {
  vehicleTypeId: "bike";
  count: number;
};
export type StationInfo = {
  stationId: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
  isVirtualStation: boolean;
  capacity: number;
};
export type StationStatus = {
  stationId: string;
  isInstalled: boolean;
  isRenting: boolean;
  isReturning: boolean;
  lastReported: number;
  numDocksAvailable: number;
  vehicleTypesAvailable: VehicleType[];
};

export type Station = {
  info: StationInfo;
  status: StationStatus;
};
